manifests/grubby.pp

```diff
@@ -0,0 +1,21 @@
+# Private class.
+class kernel::grubby{
+  if $caller_module_name != $module_name {
+    fail("Use of private class ${name} by ${caller_module_name}")
+  }
+
+  $grub_default_kernel = $kernel::grub_default_kernel ? {
+    'UNSET' => $kernel::kernel_version,
+    default => $kernel::grub_default_kernel,
+  }
+
+  # Under what circumstances could $grub_default_kernel *still*  == 'UNSET'?
+  if $kernel::set_default_kernel and $grub_default_kernel != 'UNSET' {
+    exec { 'set default kernel':
+      command => "/sbin/grubby --set-default=/boot/vmlinuz-${grub_default_kernel}",
+      path    => ['/bin','/usr/bin'],
+      unless  => "echo `/sbin/grubby --default-kernel` | grep -q ${grub_default_kernel}",
```

Comment:
Should probably be more strict about the `unless` line.  Maybe set a private variable like `$_default_kernel = /boot/vmlinux-${grub_default_kernel}` and reuse it in `--set-default` and in the `grep` portion.  Also the echo and backticks I don't think are necessary.

```
# grubby --default-kernel | grep -q /boot/vmlinuz-2.6.32-504.el6.x86_64
# echo $?
0
# grubby --default-kernel | grep -q /boot/vmlinuz-2.6.32-504.8.1.el6.x86_64
# echo $?
1
```